content/how-to/bpmn/debug-workflows.md

```diff
@@ -0,0 +1,248 @@
+width="80%"
+>}}
+<!-- vale on -->
+
+### JSON schema validation
+
+Validation can greatly limit the scope of possible errors.
+To validate your JSON payloads, use the [JSON schema task]({{< relref "/how-to/bpmn/bpmn-elements#json-schema" >}}).
+
+The JSON schema task outputs a boolean value that indicates whether the input conforms to the schema that you set.
+You can then set a condition based on whether this `valid` variable is true, and create logic to handle errors accordingly.
+For example, this schema requires that the input variables include a property `arr` whose
+value  is an array of numbers.
+
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Generated schema for Root",
+  "type": "object",
+  "properties": { "arr": { "type": "array", "items": { "type": "number" } } },
+  "required": ["arr"]
+}
+```
+
+In a production workflow, you might use this exact schema to validate the input for a function that calculates statistics (perhaps choosing a different variable name).
+
+
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-json-schema.png"
+alt="Screenshot of a conditional that branches when the JSON schema task receives invalid input."
+caption="A conditional that branches when the JSON schema task receives invalid input. [Download the template](https://github.com/libremfg/bpmn-templates/tree/main/call-activity-calculate-stats)"
+width="35%"
+>}}
+
+
+
+
+### JSONata conditions
+
+Besides the logical gateway, it may make sense to use JSONata ternary expressions in one of the many parameters that accepts JSONata expressions.
+For example, this expression creates one message body if `valid` is `true` and another if not:
+
+```jsonata
+=
+{
+  "message": $.valid ? "Schema is valid" : "Invalid schema" 
+}
+```
+
+### Check JSONata output
+
+If a field has no value, JSONata outputs nothing.
+For example, the following expression outputs only `{"name": "Rhize"}`,
+because no `$err` field exists.
+
+{{< tabs >}}
+{{% tab  "Expression" %}}
+```json
+=( 
+  $name := "Rhize";
+
+
+  { 
+  "name": $name,
+  "error": $err
+  }
+)
+```
+{{% /tab %}}
+{{% tab  "Output" %}}
+```
+{
+  "name": "Rhize"
+}
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+You can use this behavior to direct flows.
+For example, an exclusive gateway may have a condition such as `$exists(err)` that flows into an error-handling condition.
+
+### Define error boundary
+
+The error boundary is an intermediate event that evaluates whether a task had an error.
+To use an error boundary:
+1. Attach an intermediate event to a task.
+1. Select the wrench icon and choose the error boundary.
+1. Attach an arrow to the event that directs flows if the node returns an error.
+
+For example, this error boundary directs the control flow when a REST task returns an unexpected result.
+
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-error-boundary.png"
+alt="Screenshot of an error boundary that happens when a REST task returns a non-200 response."
```

Comment:
This example doesn't actually work. The error boundary should only trigger or the node returning an `error` in Go land. A REST task that return 404 will simply set a `responseCode: 404` in it's variable context and happily continue executing.



An example of a rest task that would follow an error boundary would be one which tries to hit a server that is currently offline. That request will hang for `timeout` trying to connect then return an `error` that the server was unreachable.

Reply:
Thanks. Given this point and the part about parallel boundaries, I think I'm just going to leave out error boundaries for v1 of this doc. I want to make sure I can document the correct behavior.

I've made an issue to keep track.

https://github.com/libremfg/rhize-docs/issues/123